backend/app/user_handler.go

```diff
@@ -784,14 +722,24 @@ func (h *Handler) RedeemVoucherHandler(c *gin.Context) {
 		return
 	}
 
-	err = internal.TransferTFTs(h.substrateClient, uint64(voucher.Value), user.Mnemonic, h.config.SystemAccount.Mnemonic)
+	wf, err := h.ewfEngine.NewWorkflow("redeem-voucher")
```

Comment:
what if transfer failed? I think we should update user balance if transfer succeeded only